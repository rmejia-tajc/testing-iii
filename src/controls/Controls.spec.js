// Test away!

import React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Controls from './Controls.js'




describe('<Controls /> Component', () => {


    it("renders without crashing", () => {
  
      const sanityCheck = render(<Controls />);
    });


    it('should match snapshot', () => {

        const tree = renderer.create(<Controls />);

        expect(tree.toJSON()).toMatchSnapshot();
    });


    it('should default to (lock gate) and (close gate)', () => {

        const { getByText } = render(<Controls />);

        const btnLock = getByText(/lock gate/i);
        const btnClose= getByText(/close gate/i);

        expect(btnLock).toBeInTheDocument();
        expect(btnClose).toBeInTheDocument();
    });


});