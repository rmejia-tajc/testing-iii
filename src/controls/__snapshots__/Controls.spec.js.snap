// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Component should match snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lockUnlockGate"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="closeOpenGate"
  >
    Close Gate
  </button>
</div>
`;
